www/js/lib/util.js

```diff
@@ -249,37 +249,63 @@ define([], function() {
         });
     }
 
-    /**
-     * Converts a Base64 Content to a Blob
-     * From https://stackoverflow.com/questions/16245767/creating-a-blob-from-a-base64-string-in-javascript
-     *
-     * @param {String} b64Data Base64-encoded data
-     * @param {String} contentType
-     * @param {Integer} sliceSize
-     * @returns {Blob}
-     */
-    function b64toBlob(b64Data, contentType, sliceSize) {
-        contentType = contentType || '';
-        sliceSize = sliceSize || 512;
-
-        var byteCharacters = atob(b64Data);
-        var byteArrays = [];
+    /*\
+    |*|
+    |*|  Base64 / binary data / UTF-8 strings utilities
+    |*|
+    |*|  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Base64_encoding_and_decoding
+    |*|
+    \*/
 
-        for (var offset = 0; offset < byteCharacters.length; offset += sliceSize) {
-            var slice = byteCharacters.slice(offset, offset + sliceSize);
-
-            var byteNumbers = new Array(slice.length);
-            for (var i = 0; i < slice.length; i++) {
-                byteNumbers[i] = slice.charCodeAt(i);
+    /* Array of bytes to Base64 string decoding */
+    function b64ToUint6 (nChr) {
+        return nChr > 64 && nChr < 91 ?
+            nChr - 65
+        : nChr > 96 && nChr < 123 ?
+            nChr - 71
+        : nChr > 47 && nChr < 58 ?
+            nChr + 4
+        : nChr === 43 ?
+            62
+        : nChr === 47 ?
+            63
+        :
+            0;
+    }
+    
+    function base64DecToArr(sBase64, nBlocksSize) {
+        var sB64Enc = sBase64.replace(/[^A-Za-z0-9+/]/g, ""), nInLen = sB64Enc.length,
+            nOutLen = nBlocksSize ? Math.ceil((nInLen * 3 + 1 >> 2) / nBlocksSize) * nBlocksSize : nInLen * 3 + 1 >> 2, taBytes = new Uint8Array(nOutLen);
+        for (var nMod3, nMod4, nUint24 = 0, nOutIdx = 0, nInIdx = 0; nInIdx < nInLen; nInIdx++) {
+            nMod4 = nInIdx & 3;
+            nUint24 |= b64ToUint6(sB64Enc.charCodeAt(nInIdx)) << 6 * (3 - nMod4);
+            if (nMod4 === 3 || nInLen - nInIdx === 1) {
+                for (nMod3 = 0; nMod3 < 3 && nOutIdx < nOutLen; nMod3++, nOutIdx++) {
+                    taBytes[nOutIdx] = nUint24 >>> (16 >>> nMod3 & 24) & 255;
+                }
+                nUint24 = 0;
             }
-
-            var byteArray = new Uint8Array(byteNumbers);
-
-            byteArrays.push(byteArray);
         }
+        return taBytes;
+    }
 
-        var blob = new Blob(byteArrays, {type: contentType});
-        return blob;
+    /**
+     Converts a dataURI to Uint8Array
+     * @param {String} dataURI The data URI to convert
+     * @returns {Uint8Array} A Uint8Array with the converted buffer
+     */
+     function dataURItoUint8Array(dataURI) {
```

Comment:
If unused, this function could be removed (and the 2 other ones above, too)

Reply:
OK. I can remove these.